packages/medusa/src/services/discount.ts

```diff
@@ -728,12 +734,23 @@ class DiscountService extends TransactionBaseService {
                 `Discount ${disc.code} is not valid for customer`
               )
             }
+          } else if (this.hasCustomersGroupCondition(disc)) {
```

Comment:
suggesstion: Should we check that at the top as one of the first checks?

Reply:
suggestion: It could be part of `canApplyForCustomer` function as the validation is within the same context.

Reply:
@adrien2p Remember to use the comments review when suggesting changes.

Reply:
I ve approved but needs to resolve your comment first

Reply:
so that it is not blocked

Reply:
I am referring to this one 😄 
<img width="705" alt="CleanShot 2023-04-11 at 12 13 59@2x" src="https://user-images.githubusercontent.com/59018053/231130397-1df1022b-401f-404e-a873-6250ae8b6914.png">

Reply:
I know, I am just saying that following your comment I ve approved it

Reply:
Sounds good @adrien2p will update as you suggested :)